src/Frame/VectorFieldCalculator.cc

```diff
@@ -12,7 +12,7 @@
 
 namespace carta {
 
-bool VectorFieldCalculator::DoCalculations(VectorFieldCallback& callback) {
+bool DoVectorFieldCalculation(Frame* frame, const std::function<void(CARTA::VectorOverlayTileData&)>& callback) {
     if (!(frame && frame->IsValid())) {
```

Comment:
Since Frame calls this function, Frame should do any needed checks (call a check function in VectorFieldCalculator), lock the mutex, then pass in anything needed in DoVectorFieldCalculation such as VectorFieldSettings, instead of passing in itself.

Reply:
Does it mean moving the code block in `DoVectorFieldCalculation` into Frame's function `CalculateVectorField`?

Reply:
I was thinking in `Frame::CalculateVectorField`:
* if frame is not valid return false
* if vector_field_settings are ok: lock mutex, call `DoVectorFieldCalculation(vector_field_settings, ...)`, unlock mutex, return result
* if vector_field_settings are not ok: clear settings, set response message, return true

I may have missed some details, but this is the idea.

Reply:
I don't think you need a check function in VectorFieldCalculator just to check the settings (stokes_intensity and stokes_angle), unless there is more to it than that